### PR TITLE
WT-10331 Split make-check-test into multiple parts

### PR DESCRIPTION
**Pull request by Sean04**
I've split the csuite test for the msan evergreen variant by adding a new ctest label for the tests that take a long time under MSAN when compiler optimisations are turned off. I've also increased the distro size for these tasks. Task times for MSAN are now around 1hour 30mins for the make-check-test and ~4 hours for the new csuite-long-running task. I've also changed the optimisation for the UBSAN variant back to -O0 without splitting the tasks as this didn't cause as much of a significant slowdown in my testing.

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -5875,6 +5885,10 @@ buildvariants:
     - name: examples-c-production-disable-static-test
     - name: format-stress-pull-request-test
     - name: make-check-test
+      distros: ubuntu2004-large
+    - name: csuite-long-running
+      distros: ubuntu2004-large
+      exec_timeout_secs: 21600 # 6 hrs
```
Comment (lukech) [inline test/evergreen.yml:5891]: Should we consider lower the running frequency for the long csuite tests (using `batchtime` setting)?

File: test/evergreen.yml [hunk 2]
```diff
@@ -2618,6 +2618,14 @@ tasks:
               ${test_env_vars|} ${python_binary|python3} ../test/suite/run.py -v 4 test_prepare_hs03.py --hook timestamp
             done
 
+  - name: csuite-long-running
```
Comment (lukech) [inline test/evergreen.yml:2621]: Does the introduction of this new task mean the `make-check-test` task will not cover those long-running csuite tests? If that's the case we should check all places that call the `make-check-test` task in evergreen.yml and evaluate whether the new task is needed.

File: test/evergreen.yml (hunk 1 above)
Comment (lukech) [inline test/evergreen.yml:5891]: The project-level setting is 6 hrs (see the "Project Settings" section), making this line of setting unnecessary.

Comment (Sean04) [reply test/evergreen.yml:5891]: Ah great, I'll remove this

Comment (Sean04) [reply test/evergreen.yml:5891]: Yes, I wan wondering that but not sure how we determine what frequency to use?

File: test/evergreen.yml (hunk 2 above)
Comment (Sean04) [reply test/evergreen.yml:2621]: The `make-check-test` task still works the same and will cover all the csuite tests however the new label `long_running` allows those tests to be excluded from `make-check-test` with the `-LE long_running` ctest argument and then included in the separate task.

File: test/evergreen.yml (hunk 1 above)
Comment (lukech) [reply test/evergreen.yml:5891]: I think running it "once per day" is good enough.

File: test/evergreen.yml (hunk 2 above)
Comment (lukech) [reply test/evergreen.yml:2621]: Ahh, I see. Did not realize "LE" stands up "label-exclude" for ctest.

File: test/evergreen.yml [hunk 3]
```diff
@@ -5864,8 +5873,9 @@ buildvariants:
       WT_TOPDIR=$(git rev-parse --show-toplevel)
       WT_BUILDDIR=$WT_TOPDIR/cmake_build
       LD_LIBRARY_PATH=$WT_BUILDDIR
+      TESTUTIL_MSAN=1
     # We don't run C++ memory sanitized testing as it creates false positives.
```
Comment (lukech) [inline test/evergreen.yml:5877]: Could this comment be extended to cover why the "long_running" label is excluded?

File: test/ctest_helpers.cmake [hunk 4]
```diff
@@ -308,13 +308,13 @@ function(define_c_test)
                 CMDS ${test_cmd}
                 DIR_NAME ${C_TEST_DIR_NAME}
                 LABELS "check;csuite"
-            )
+                )
```
Comment (korteland) [inline test/ctest_helpers.cmake:311]: Nitpick, indentation here looks like it got messed up.

File: test/csuite/wt4105_large_doc_small_upd/main.c [hunk 5]
```diff
@@ -149,7 +149,8 @@ main(int argc, char *argv[])
             modify_entry.offset = offset;
             modify_entry.size = modify_entry.data.size;
             /* FIXME-WT-6113: extend timeout to pass the test */
-            (void)alarm(15);
+            if (!testutil_is_flag_set("TESTUTIL_MSAN"))
+                (void)alarm(15);
```
Comment (korteland) [inline test/csuite/wt4105_large_doc_small_upd/main.c:153]: Why do we need to avoid this for the MSan variant?

**Review by lukech — APPROVED** [review]
LGTM

File: test/evergreen.yml [hunk 6]
```diff
@@ -5852,6 +5860,7 @@ buildvariants:
   expansions:
     CMAKE_TOOLCHAIN_FILE: -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/mongodbtoolchain_stable_clang.cmake
     CMAKE_BUILD_TYPE: -DCMAKE_BUILD_TYPE=MSan
+    CC_OPTIMIZE_LEVEL: -DCC_OPTIMIZE_LEVEL=-O0
```
Comment (korteland) [inline test/evergreen.yml:5863]: I would prefer if at least one of these `-O0` optimisation levels had some explanation for why (i.e. that Clang treats `-Og` as `-O1` and makes the test binary much more difficult to debug). Unless it's already in this file somewhere?

**Review by korteland — COMMENTED** [review]
Thanks for persisting with this - this solution overall definitely feels like the "right" way to solve this.

File: test/csuite/wt4105_large_doc_small_upd/main.c (hunk 5 above)
Comment (Sean04) [reply test/csuite/wt4105_large_doc_small_upd/main.c:153]: This alarm will trigger due to msan significatnly slowing down the performance when optimisation is off. The alarm is there to catch any performance regressions but I don't think we should be concerned about that when testing with msan.

Comment (Sean04) [reply test/csuite/wt4105_large_doc_small_upd/main.c:153]: I'll add a comment.

Comment (korteland) [reply test/csuite/wt4105_large_doc_small_upd/main.c:153]: Oh I see - that makes sense now, thanks for explaining that. Yes, I agree a comment is a good idea.

File: test/evergreen.yml (hunk 6 above)
Comment (Sean04) [reply test/evergreen.yml:5863]: -

**Review by korteland — APPROVED** [review]
LGTM!

**Comment by Sean04** [comment]
evergreen merge

**Pull request merged**